classes/local/store/s3/client.php

```diff
@@ -292,35 +292,33 @@ protected function get_exception_details($exception) {
         return $details;
     }
 
-    public function define_amazon_s3_check($mform, $testdelete = true) {
-        global $OUTPUT;
-        $connection = false;
-
+    public function define_amazon_s3_check($testdelete = true) {
+        global $SESSION;
+        $SESSION->notifications = [];
```

Comment:
This feels wrong to me, wont this get run on every admin page when the admin tree is loaded?

Reply:
No, because of the check made on line 113 of settings.php which checks if the current page is the settings page. I added that check to stop it from trying to test the connection when it is on any other admin page.

Reply:
Ah see my comments here https://github.com/catalyst/moodle-tool_objectfs/pull/205/files#r364041364